### ISSUE TITLE
ui: statements page leaks sensitive information

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#44348]
tldr: the "statements page" in the admin UI and the `/_status/statements` endpoint are incorrectly available to non-admin users. This is a new variant of  #42567. Impact is, comparatively, minor.

### Impact

All deployments up to 2.1.11, 19.1.7, 19.1.6 are affected.

Affected deployments risk showing database and table *names* to users who do not have privileges over them, as well as the SQL text of queries from other users. The content of tables and the particular values in the SQL queries is not leaked.

### Details

The code in the `sql` package is pretty clear about the idea that `crdb_internal.node_statement_statistics` is meant to be used only by admin users.

The reason is that it shows database and table names that the current user may not have privilege over.

Since the data structure that retains statements stats does not memorize which user has issued which statements, it is not possible to filter out the data to only show the statements that the current user has issued. So the conservative behavior is to block non-admin access.

This block is properly implemented for the virtual table `crdb_internal.node_statement_statistics`.

Unfortunately, the code that delivers the `/_status/statement` HTTP endpoint does not use the same code path and thus does not authorize the request properly.

This is more or less a remnant of #42567. I had missed that endpoint because .. it is served in a different Go file and the final guardrails recommended in #44150 are not yet implemented.

**Comment by knz** [comment]
Fixed by #44349 and backports
